recipes-core/odhcpd/odhcpd_git.bb

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2018 Daniel Dickinson <cshored@thecshore.com>
+# Released under the MIT license (see COPYING.MIT for the terms)
+
+DESCRIPTION = "OpenWrt DHCP/DHCPv6(-PD)/RA Server & Relay"
+HOMEPAGE = "http://git.openwrt.org/?p=project/odhcpd.git;a=summary"
+LICENSE = "GPL-2.0"
+LIC_FILES_CHKSUM = "file://src/odhcpd.c;beginline=1;endline=13;md5=b5b1da01ca7e1cdd0308c552dc0a1384"
+SECTION = "base"
+DEPENDS = "libubox ubus libnl uci"
+RDEPENDS_${PN} += " base-files-scripts-openwrt "
+
+inherit cmake pkgconfig openwrt openwrt-services openwrt-base-files
+
+SRCREV_odhcpd = "750e457e3000187b85906814a2529ede24775325"
+SRCREV_openwrt = "${OPENWRT_SRCREV}"
+
+SRC_URI = "\
+    git://git.openwrt.org/project/odhcpd.git;name=odhcpd \
+"
+
+S = "${WORKDIR}/git"
+OF = "${S}/openwrt/package/network/services/odhcpd/files"
+
+EXTRA_OECMAKE_append = " -DUBUS=1 -Dlibnl=${STAGING_LIBDIR}/libnl-3.so -Dlibnl-tiny_include_dir=${STAGING_INCDIR}/libnl3"
```

Comment:
hmm hardcoded path to a library seems a bit odd. Yocto used sysroot to find target libs is that not working ?

Reply:
no it's not; problem is the recipe is called libnl but the library is libnl-3 - the PROVIDES goes off the recipe name not the library name so Yocto doesn't find nl-3 - I guess that could be considered a bug in core.

Reply:
will something like
```
-Dlibnl=-lnl-3
```

work ?

Reply:
I'll try that -- I tried other things for -Dlibnl but I guess it was the wrong syntax.

Reply:
didn't work, get build/tmp-glibc/work/armv5e-oe-linux-gnueabi/odhcpd/git-r0/build'
| make[2]: *** No rule to make target '-lnl-3', needed by 'odhcpd'.  Stop.

Looking at odhcpd's CMakeLists.txt it's kind of funky I think.  Probably need to patch it.

The issue is that upstream has libnl-tiny and that's assumption goes into CMakeLists.txt